### PR TITLE
Adding Model of BlueROV2 (heavy configuration)

### PR DESCRIPTION
**Pull request by lennartalff**
This pull request refers to the pull request [#16004](https://github.com/PX4/Firmware/pull/16004) in the PX4 Firmware repository.

This PR adds the gazebo model of the BlueROV2 in heavy configuration.

### CODE REVIEW

**Comment by TSC21** [comment]
@lennartalff cool! You mind attaching a screenshot of it? Thanks

**Comment by lennartalff** [comment]
@TSC21 Here you are!
bluerov2-1
bluerov2-2
bluerov2-3

**Comment by TSC21** [comment]
> @TSC21 Here you are!
> bluerov2-1
> bluerov2-2
> bluerov2-3

Looks awesome!

File: models/uuv_bluerov2_heavy/uuv_bluerov2_heavy.sdf [hunk 1]
```diff
@@ -0,0 +1,663 @@
...
+                <origin>0 0 0</origin>
+                <compensation>1</compensation>
+                <height_scale_limit>0.05</height_scale_limit>
+            </buoyancy>
+        </plugin>
+
+        <plugin name="mavlink_interface" filename="libgazebo_mavlink_interface.so">
+            <robotNamespace />
+            <imuSubTopic>/imu</imuSubTopic>
+            <gpsSubTopic>/gps</gpsSubTopic>
```
Comment (Jaeyoung-Lim) [inline models/uuv_bluerov2_heavy/uuv_bluerov2_heavy.sdf]:
```suggestion
```

File: models/uuv_bluerov2_heavy/uuv_bluerov2_heavy.sdf [hunk 2]
```diff
@@ -0,0 +1,663 @@
...
+            <noiseDensity>0.0004</noiseDensity>
+            <randomWalk>6.4e-06</randomWalk>
+            <biasCorrelationTime>600</biasCorrelationTime>
+            <magTopic>/mag</magTopic>
+        </plugin>
+
+        <plugin name="barometer_plugin" filename="libgazebo_barometer_plugin.so">
+            <robotNamespace />
+            <pubRate>50</pubRate>
+            <baroTopic>/baro</baroTopic>
```
Comment (Jaeyoung-Lim) [inline models/uuv_bluerov2_heavy/uuv_bluerov2_heavy.sdf:527]: Just curious, for UUVs, the barometer model here will not match the barometer on the real vehicle right? How does this usually influence the state estimation on the vehicle? Do we need to make it constant?

**Review by Jaeyoung-Lim — CHANGES_REQUESTED** [review]
Thanks! This is really awesome!

I have just one minor question
- Does the `heavy` suffix imply that there are other configurations that are planned to be added? If not, would it make sense to simplify the namespace by removing the `heavy` suffix?

File: models/uuv_bluerov2_heavy/uuv_bluerov2_heavy.sdf (hunk 2 above)
Comment (lennartalff) [reply models/uuv_bluerov2_heavy/uuv_bluerov2_heavy.sdf:527]: In fact we currently use external vision estimation for the height. So the ekf2 ignores the input of the baro. But in the future i would like to modify the pressure->height calculation in `VehicleAirdata.cpp` https://github.com/PX4/Firmware/blob/0b74076265edda2ac34cb9c4addcfeea7277d3a2/src/modules/sensors/vehicle_air_data/VehicleAirData.cpp#L245 to check if the vehicle type is a submarine and adapt the equation accordingly.

Then i would also modify the barometric pressure simulation accordingly to get reasonable results.

**Comment by lennartalff** [comment]
@Jaeyoung-Lim in fact the "heavy" suffix means that this configuration uses 8 thrusters. The standard configuration only has 6. So in case I or someone else will add the standard configuration for this vehicle no confusing renaming of the models is needed. So i think it would be a good idea to keep the suffix.

**Review by Jaeyoung-Lim — APPROVED** [review]
Thanks!

Is there anymore outstanding issues? Could you make it as ready to review?

**Comment by hamishwillee** [comment]
Docs? 
Normally in table here https://dev.px4.io/master/en/simulation/gazebo.html#running-the-simulation and with image here: https://dev.px4.io/master/en/simulation/gazebo_vehicles.html 

If there is an associated "world" then here: https://dev.px4.io/master/en/simulation/gazebo_worlds.html

**Pull request merged**